### ISSUE TITLE
Relationship between toml and json config files

### ISSUE THREAD

**Issue by user-0209** [bmaltais/kohya_ss#2516]
Amongst all the documentation I don't seem to find clarity about the purpose/function of toml configuration files, json configuration files and commandline args. Would be good, if some expert could shed light on this in terms of settling those many parameters in a defined way. Thanks.

**Comment by oO0** [comment]
Why not to keep settings in single file?

**Comment by b-fission** [comment]
I'm not an expert at this, but I'll try. Currently, there are two kinds of toml files:

### Script arguments toml
This type of toml contains the parameters or options that control the training process and its output. Options like learn-rate, rank, alpha, scheduler, optimizer, noise type, file paths, precision, and general dataset configs.

In the past version of the gui, parameters would've been passed via command-line arguments, meaning this type of toml was not used in version 23.x or earlier. Now as of version 24.x of the gui, this type of toml file is autogenerated whenever the gui runs a script or begins training, and contains the arguments for the process that's being run.

The [given reason](https://github.com/bmaltais/kohya_ss/releases/tag/v24.0.1) for the use of toml instead of full command-lines is better security from malicious config arguments.

### Dataset config toml
This type of toml describes how your input files (image and captions) should be handled. A lot of documentation on this type of file is given here: https://github.com/kohya-ss/sd-scripts/blob/main/docs/config_README-en.md

Personally I haven't found it necessary to create or edit this type of toml, but it could be useful for cases where you're training multiple concepts with several images folders, and want to override certain options on a specific folder.

### Gui config json

This type of file contains the options that are configurable in the gui. You can think of it as your project file, which you can use to save and load your settings in the gui. The json config is only used by the gui and not passed to the training scripts. A copy of the config json is autogenerated when you start training.

**Comment by bmaltais** [comment]
I would not have said it better!
